File: sbc/Maratona 2024/k.py
MOD = 998244353

def mul_inv(a):
    return pow(a, MOD - 2, MOD)

def prob(T, K):
    num = den = pow(2, 1000000, MOD)
    for i in range(1, 1 << K):
        p = cnt = 1
        for j in range(K):
            if i & (1 << j):
                p = (p * (j + 1)) % MOD
                cnt += 1
        if cnt % 2 == K % 2:
            num = (num + den // p) % MOD
        else:
            num = (num - den // p) % MOD
    num = (num * pow(T, K - 1, MOD)) % MOD
    prob = (num * mul_inv(den)) % MOD
    return (prob * mul_inv(den - 1)) % MOD

T, K = map(int, input().split())
print(prob(T, K))

File: sbc/Maratona 2024/k.c
#include <stdio.h>
#include <math.h>
#define MOD 998244353

long long mul_inv(long long a) {
    return pow(a, MOD - 2) % MOD;
}

long long prob(long long T, long long K) {
    long long num = pow(T, K - 1) % MOD;
    for (long long p = 2; p <= K; p++) {
        num = (num * (p * T - 1) % MOD * mul_inv(p)) % MOD;
    }
    long long den = pow(2, 1000000) % MOD;
    long long prob = (num * mul_inv(den)) % MOD;
    return (prob * mul_inv(den - 1)) % MOD;
}

int main() {
    long long T, K;
    scanf("%lld %lld", &T, &K);
    printf("%lld\n", prob(T, K));
    return 0;
}
